annotations/static/annotations/js/annotators/appellationlist.js

```diff
@@ -1,129 +1,265 @@
+                );
+            }
         },
         watchCheckStore: function () {
-            store.watch(
-                (state) => {
-                    return store.getters.getSelect
-                },
-                (val) => {
-                    if (val) {
-                        this.checkAll()
-                    }
-                },
-            );
-        },
-
-        uncheckAll: function () {
-            this.checked = false;
-        },
-        checkAll: function () {
-            this.checked = true;
-        },
-        hide: function () {
-            this.$emit("hideappellation", this.appellation);
-        },
-        show: function () {
-            this.$emit("showappellation", this.appellation);
-        },
-        toggle: function () {
-            if (this.appellation.visible) {
-                this.hide();
-            } else {
-                this.show();
+            if (typeof store !== 'undefined' && store.watch) {
+                store.watch(
+                    (state) => { return store.getters.getSelect; },
+                    (val) => { if (val) { this.checkAll(); } }
+                );
             }
         },
-        isSelected: function () {
-            return this.appellation.selected;
-        },
-        select: function () {
-            this.$emit('selectappellation', this.appellation);
+        uncheckAll: function () { this.checked = false; },
+        checkAll: function () { this.checked = true; },
+        hide: function () { this.$emit("hideappellation", this.appellation); },
+        show: function () { this.$emit("showappellation", this.appellation); },
+        toggle: function () {
+            if (this.appellation.visible) { this.hide(); } else { this.show(); }
         },
+        isSelected: function () { return this.appellation.selected; },
+        select: function () { this.$emit('selectappellation', this.appellation); },
         label: function () {
-            if (this.appellation.interpretation) {
+            if (this.appellation && this.appellation.interpretation) {
                 return this.appellation.interpretation.label;
-            } else if (this.appellation.dateRepresentation) {
+            } else if (this.appellation && this.appellation.dateRepresentation) {
                 return this.appellation.dateRepresentation;
             }
+            return ''; 
         },
         getCreatorName: function (creator) {
-            if (creator.id == USER_ID) {
+            if (creator && typeof USER_ID !== 'undefined' && creator.id == USER_ID) {
                 return 'you';
-            } else {
+            } else if (creator && creator.username) {
                 return creator.username;
             }
+            return 'Unknown';
         },
         getFormattedDate: function (isodate) {
-            return moment(isodate).format('dddd LL [at] LT');
-        }
+            if (typeof moment !== 'undefined') {
```

Comment:
when would this happen? wouldn't this be something that can be avoided by ensuring `moment` is included?